### ISSUE TITLE
Convert to using NX Monorepo

### ISSUE THREAD

**Issue by WonderPanda** [golevelup/nestjs#143]

**Comment by underfisk** [comment]
Alongside with Nx i would suggest `pnpm`, i can have a PR soon for this after we release the next version that supports nestjs v8

**Comment by WonderPanda** [comment]
The biggest thing that I don't have a clear understanding of yet is how we should implement the publishing step with NX. This is currently something that Lerna still does much better as it takes care of incrementing all the version numbers for dependent packages and releasing them together.

Maybe @jmcdo29 has this all figured out now in his Ogma repo?

**Comment by underfisk** [comment]
He has, he's using https://github.com/changesets/changesets which is great

**Comment by WonderPanda** [comment]
Cool, does it have an NX plugin integration? Or what is the rough process we would have to follow to integrate it?

**Comment by underfisk** [comment]
https://dev.to/jmcdo29/automating-your-package-deployment-in-an-nx-monorepo-with-changeset-4em8

**Comment by WonderPanda** [comment]
I have also seen this NX Plugin advertised by the official NX twitter account which would be worth looking in to:
https://github.com/Dev-Thought/nx-plugins/tree/master/libs/nx-deploy-it

**Comment by WonderPanda** [comment]
I've seen this package mentioned as a good option on the official Nrwl Twitter account: https://github.com/jscutlery/semver

Would be interesting to see what the pros and cons are of this versus a manual setup using changesets

I like that this one is an actual NX plugin as opposed to a lot of glue scripts tying everything together

**Comment by jeremylvln** [comment]
Oh, I started implementing this a lot of time ago but didn't manage to finish. I've done it today.
Unfortunately, I didn't see the discussion here beforehand, sorry about that. I've opened a PR to migrate to Nx, but I indeed need to work on the publishing step.

**EDIT:** I've added the `semver` plugin and plugged the npm publish integration too. The implementation seems fine, but it should be tested after merging to confirm :)
